### PR TITLE
Implement state machine

### LINKED ISSUE #13: Implement state machine model
Add the ability to detect a sensor malfunction when there is an impossible(during normal operation) shift in sensor states.

### PR DESCRIPTION
**Pull request by ashroom1**

### CODE REVIEW

File: TankControl.ino [hunk 1]
```diff
@@ -413,6 +433,20 @@ void loop() {
     s2 = !digitalRead(SENSOR2);
     s3 = !digitalRead(SENSOR3);
 
+    if(((!s3prev && !s2prev && !s1prev) && (!s3 && s2 && !s1)) || ((!s3prev && !s2prev) || (!s3 && s2))) {
```
Comment (ashroom1) [inline TankControl.ino:436]: ((!s3prev && !s2prev && !s1prev) && (!s3 && !s2 && s1))

Comment (ashroom1) [reply TankControl.ino:436]: Switch debouncing from sensor to be considered.

Comment (LikithBC) [reply TankControl.ino:436]: Cannot be implemented because of hysteresis (sensor) issue.

**Comment by LikithBC** [comment]
Cannot be implemented because of hysteresis (sensor) issue.

**Pull request closed without merging**